cfgov/privacy/forms.py

```diff
@@ -0,0 +1,212 @@
+        total = 0
+        for f in files:
+            total += f.size
+        return total
+
+    def limit_file_size(self, files):
+        mb = 1048576  # one megabyte in bytes
+        max_bytes = mb * 6  # 6 MB
+        total_uploaded_bytes = self.combined_file_size(files)
+        if total_uploaded_bytes > max_bytes:
+            display_size = round(total_uploaded_bytes / mb, 1)
+            err = forms.ValidationError(
+                f"Total size of uploaded files ({display_size} MB) was "
+                "greater than size limit (2 MB)."
+            )
+            self.add_error('supporting_documentation', err)
+
+    def limit_number_of_files(self, files):
+        max_files = 6
+        if len(files) > max_files:
+            err = forms.ValidationError(
+                f"Please choose {max_files} or fewer files. "
+                f"You chose {len(files)}."
+            )
+            self.add_error('supporting_documentation', err)
+
+    def clean(self):
+        super().clean()
+        self.require_address_if_mailing()
+        uploaded_files = self.files.getlist('supporting_documentation')
+        self.limit_file_size(uploaded_files)
+        self.limit_number_of_files(uploaded_files)
+
+    # Email message
+    def send_email(self):
+        uploaded_files = self.files.getlist('supporting_documentation')
+        data = self.cleaned_data
+        data.update({'uploaded_files': uploaded_files})
+        subject = self.format_subject(data['requestor_name'])
+        from_email = settings.DEFAULT_FROM_EMAIL
+        recipient_list = ['FOIA@consumerfinance.gov']
+
+        body = self.email_body(data)
+
+        email = EmailMessage(
+            subject,
+            body,
+            from_email,
+            recipient_list,
+            reply_to=[data['requestor_email']]
+        )
+        email.content_subtype = 'html'
+
+        for f in uploaded_files:
+            email.attach(f.name, f.read(), f.content_type)
+
+        try:
+            email.send()
+        except BadHeaderError:
+            return HttpResponse('Invalid header found.')
+
+
+class DisclosureConsentForm(PrivacyActForm):
+    consent = forms.BooleanField(
+        widget=forms.CheckboxInput(attrs={'class': 'a-checkbox'}),
+    )
+    # Additional fields beyond what's defined in PrivacyActForm
+    recipient_name = forms.CharField(
+        label='Name of recipient',
+        widget=forms.TextInput(attrs=text_input_attrs),
+    )
+    recipient_email = forms.EmailField(
+        label='Email address',
+        widget=forms.EmailInput(attrs=text_input_attrs),
+    )
+
+    def format_subject(self, name):
+        return f'Disclosure request from consumerfinance.gov: {name}'
+
+    email_template = 'privacy/disclosure_consent_email.html'
+
+    def email_body(self, data):
+        num_files = len(data['uploaded_files'])
+        data.update({'num_files': num_files,
+                     'consent_text': self.consent_text})
```

Comment:
I get a 500 error when I try to submit this form (at /privacy/disclosure-consent/) because `self.consent_text` isn't defined here, but only on RecordsAccessForm.

Reply:
Ack. That's a clash between my code and Wyatt's. Thanks for that. Fix on the way.

Reply:
Fixed in 79e192b15158ce8b9b76b4542fa619a2649fbcca, I think.